src/thread_pool/thread_pool.ml

```diff
@@ -0,0 +1,75 @@
+open Stdune
+
+type t =
+  { mutex : Mutex.t
+  ; cv : Condition.t
+  ; spawn_thread : (unit -> unit) -> unit
+  ; tasks : (unit -> unit) Queue.t
+  ; min_workers : int
+  ; max_workers : int
+  ; (* number of threads waiting for a task *)
+    mutable idle : int
+  ; (* total number of running threads *)
+    mutable running : int
+  ; mutable dead : Thread.t list
```

Comment:
I'm surprised that the previous field has a comment while this one, which seems far less obvious, doesn't.

Reply:
I'll add a comment. It's simple enough really, when a thread terminates, we need to `Thread.join` it to make the OS release all the resources. Usually, it doesn't matter, but here we have a dynamic number of threads so we don't want to leak memory.